back/app/services/side_fx_initiative_status_change_service.rb

```diff
@@ -6,6 +6,16 @@ class SideFxInitiativeStatusChangeService
   def before_create(change, user); end
 
   def after_create(change, user)
+    lock_initiative_editing_if_required(change)
+
     InitiativeStatusService.new.log_status_change change, user: user
   end
+
+  private
+
+  def lock_initiative_editing_if_required(change)
+    return unless change.initiative_status.code == 'proposed' && Initiative.review_required?
```

Comment:
Consider this scenario:
1. There is a proposal.
2. They turn on the feature.
3. The author edits that proposal.

Result: the proposal gets locked. Should it happen? Probably, not. Maybe it makes sense to lock it only at the moment when the status actually changes.

```suggestion
    return unless change.initiative_status.code == 'proposed' && change.initiative_status_changed? && Initiative.review_required?
```
(not sure about the exact method. If it doesn't work check https://api.rubyonrails.org/v5.1/classes/ActiveRecord/AttributeMethods/Dirty.html)

Reply:
Good thinking!

Reply:
@alexander-cit Not sure I understand...

This code is in `side_fx_status_change_service` and is only called after the `initiative_status_change` is created and only updates the attribute if the status the initiative was changed to is 'proposed'.

Why do you think editing the proposal when the feature is on will result in editing being locked? (It will only get locked when feature is on when it changes to proposed status... and in this case it already has, before feature was turned on)

Reply:
@alexander-cit Not sure I understand...

This code is in side_fx_status_change_service and is only called after the initiative_status_change is created and only updates the attribute if the status the initiative was changed to is 'proposed'.

Why do you think editing the proposal when the feature is on will result in editing being locked? (It will only get locked when feature is on when it changes to proposed status... and in this case it already has, before feature was turned on)

Reply:
i.e. editing an initiative does not create an `initiative_status_change`

Reply:
> only called after the initiative_status_change is created

Ah indeed. Then you can ignore this comment.